### PR TITLE
add SameSite attribute for SSO cookie metatype

### PR DESCRIPTION
**Pull request by utle**
#10808

### CODE REVIEW

**Comment by utle** [comment]
#build

**Comment by utle** [comment]
#build

**Comment by utle** [comment]
#build

**Comment by utle** [comment]
#run-libby-bot

**Comment by pnicolucci** [comment]
@dmuelle  the text SHOULD be different. Security vs session two different sets of cookies/config that we're working with.

**Comment by utle** [comment]
#build

File: dev/com.ibm.ws.webcontainer.security.app/resources/OSGI-INF/l10n/metatype.properties [hunk 1]
```diff
@@ -101,3 +101,11 @@ contextRootForFormAuthenticationMechanism.desc=Specifies the context root of a f
 
 basicAuthenticationMechanismRealmName=Realm name of the Java EE 8 basic authentication
 basicAuthenticationMechanismRealmName.desc=Specifies a realm name for the Java EE 8 Security HTTP basic authentication. This value is applicable when overrideHttpAuthMethod is set to BASIC, or if overrideHttpAuthMethod is set to CLIENT_CERT and allowAuthenticationFailOverToAuthMethod is set to BASIC. The default value is defaultRealm.
+
+# The word SameSite should not be translated
+sameSiteCookie=Add the SameSite attribute to the SSO cookie
+sameSiteCookie.desc=Specifies the SameSite attribute value to use for SSO cookie.
```
Comment (dmuelle) [inline dev/com.ibm.ws.webcontainer.security.app/resources/OSGI-INF/l10n/metatype.properties]: **change**
Specifies the SameSite attribute value to use for SSO cookie.
**to**
Specifies the SameSite attribute value to use for SSO cookies.
**or**
Specifies the SameSite attribute value to use for the SSO cookie.

**Comment by skasund** [comment]
L2 message review not required

File: dev/com.ibm.ws.webcontainer.security.app/resources/OSGI-INF/l10n/metatype.properties (hunk 1 above)
Comment (utle) [reply dev/com.ibm.ws.webcontainer.security.app/resources/OSGI-INF/l10n/metatype.properties]: Done.
Specifies the SameSite attribute value to use for the SSO cookie.

**Comment by utle** [comment]
#build

File: dev/com.ibm.ws.webcontainer.security.app/resources/OSGI-INF/l10n/metatype.properties [hunk 2]
```diff
@@ -101,3 +101,11 @@ contextRootForFormAuthenticationMechanism.desc=Specifies the context root of a f
 
 basicAuthenticationMechanismRealmName=Realm name of the Java EE 8 basic authentication
 basicAuthenticationMechanismRealmName.desc=Specifies a realm name for the Java EE 8 Security HTTP basic authentication. This value is applicable when overrideHttpAuthMethod is set to BASIC, or if overrideHttpAuthMethod is set to CLIENT_CERT and allowAuthenticationFailOverToAuthMethod is set to BASIC. The default value is defaultRealm.
+
+# The word SameSite should not be translated
+sameSiteCookie=Add the SameSite attribute to the SSO cookie
```
Comment (dmuelle) [inline dev/com.ibm.ws.webcontainer.security.app/resources/OSGI-INF/l10n/metatype.properties]: add period
Add the SameSite attribute to the SSO cookie.

File: dev/com.ibm.ws.webcontainer.security.app/resources/OSGI-INF/l10n/metatype.properties [hunk 3]
```diff
@@ -101,3 +101,11 @@ contextRootForFormAuthenticationMechanism.desc=Specifies the context root of a f
...
+sameSiteCookie.desc=Specifies the SameSite attribute value to use for the SSO cookie.
```
Comment (dmuelle) [inline dev/com.ibm.ws.webcontainer.security.app/resources/OSGI-INF/l10n/metatype.properties]: Specifies the SameSite attribute value to use for the SSO cookie.

**change to**
Specify the SameSite attribute value to use for the SSO cookie.

File: dev/com.ibm.ws.webcontainer.security.app/resources/OSGI-INF/l10n/metatype.properties [hunk 4]
```diff
@@ -101,3 +101,11 @@ contextRootForFormAuthenticationMechanism.desc=Specifies the context root of a f
...
+sameSiteCookieDisabled=Do not set the SameSite attribute value on the SSO cookie
```
Comment (dmuelle) [inline dev/com.ibm.ws.webcontainer.security.app/resources/OSGI-INF/l10n/metatype.properties]: add period

File: dev/com.ibm.ws.webcontainer.security.app/resources/OSGI-INF/l10n/metatype.properties [hunk 5]
```diff
@@ -101,3 +101,11 @@ contextRootForFormAuthenticationMechanism.desc=Specifies the context root of a f
...
+sameSiteCookielax=Set the SSO cookie SameSite attribute value to Lax
```
Comment (dmuelle) [inline dev/com.ibm.ws.webcontainer.security.app/resources/OSGI-INF/l10n/metatype.properties]: add period

File: dev/com.ibm.ws.webcontainer.security.app/resources/OSGI-INF/l10n/metatype.properties [hunk 6]
```diff
@@ -101,3 +101,11 @@ contextRootForFormAuthenticationMechanism.desc=Specifies the context root of a f
...
+sameSiteCookieNone=Set the SSO cookie SameSite attribute value to None
```
Comment (dmuelle) [inline dev/com.ibm.ws.webcontainer.security.app/resources/OSGI-INF/l10n/metatype.properties]: add period

File: dev/com.ibm.ws.webcontainer.security.app/resources/OSGI-INF/l10n/metatype.properties [hunk 7]
```diff
@@ -101,3 +101,11 @@ contextRootForFormAuthenticationMechanism.desc=Specifies the context root of a f
...
 basicAuthenticationMechanismRealmName=Realm name of the Java EE 8 basic authentication
 basicAuthenticationMechanismRealmName.desc=Specifies a realm name for the Java EE 8 Security HTTP basic authentication. This value is applicable when overrideHttpAuthMethod is set to BASIC, or if overrideHttpAuthMethod is set to CLIENT_CERT and allowAuthenticationFailOverToAuthMethod is set to BASIC. The default value is defaultRealm.
+
+# The word SameSite should not be translated
+sameSiteCookie=Add the SameSite attribute to the SSO cookie
+sameSiteCookie.desc=Specifies the SameSite attribute value to use for the SSO cookie.
+sameSiteCookieDisabled=Do not set the SameSite attribute value on the SSO cookie
+sameSiteCookielax=Set the SSO cookie SameSite attribute value to Lax
+sameSiteCookieNone=Set the SSO cookie SameSite attribute value to None 
+sameSiteCookieStrict=Set the SSO cookie SameSite attribute value to Strict
```
Comment (dmuelle) [inline dev/com.ibm.ws.webcontainer.security.app/resources/OSGI-INF/l10n/metatype.properties]: add period

**Comment by utle** [comment]
#run-libby-bot

**Pull request merged**